tests/test_deprecated.py

```diff
@@ -0,0 +1,43 @@
+"""Unittests for deprecated features"""
+
+
+import affine
+import pytest
+
+import rasterio
+from rasterio.transform import guard_transform
+
+
+def test_open_affine_and_transform(path_rgb_byte_tif):
+    with pytest.raises(ValueError) as e:
+        with rasterio.open(
+                path_rgb_byte_tif,
+                affine=affine.Affine.identity(),
+                transform=affine.Affine.identity()) as src:
+            pass
+        for word in 'affine', 'transform', 'exclusive':
+            assert word in str(e)
+
+
+def test_open_transform_not_Affine(path_rgb_byte_tif):
+    with pytest.raises(TypeError):
+        with rasterio.open(
+                path_rgb_byte_tif,
+                transform=tuple(affine.Affine.identity())) as src:
+            pass
+
```

Comment:
This inline comment is still showing as unchanged in the GH conversation, but I addressed it in the most recent batch of commits.